doc/tutorial.md

```diff
@@ -408,7 +408,16 @@ give you the idea:
 | `A => bar(<>)`       | `<a:A> => bar(a)`          |
 | `A B => bar(<>)`     | `<a:A> <b:B> => bar(a, b)` |
 | `<A> B => bar(<>)`   | `<a:A> B => bar(a)`        |
+| `<p:A> B => bar(<>)` | `<p:A> B => bar(p)`        |
 | `<A> <B> => bar(<>)` | `<a:A> <b:B> => bar(a, b)` |
+| `<p:A> <q:B> => bar(<>)` | `<p:A> <q:B> => bar(p, q)` |
+| `<p:A> B => Foo{<>}` | `<p:A> B => Foo{p:p}` |
+| `<p:A> <q:B> => Foo{<>} | `<p:A> <q:B> => Foo{p:p, q:q}` |
+
+The funky expression also works with struct constructors 
+(like `Foo{...}` in examples above). In such case user had to explicitly
```

Comment:
Nit: Instead of "In such case user had to explicitly...", maybe "This works out well if the names of your parsed values match the names of your struct fields."